apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.html.ts

```diff
@@ -65,23 +67,86 @@ export function StudentAssessmentInstance({
         ${compiledScriptTag('studentAssessmentInstanceClient.ts')}
       </head>
       <body>
+        ${renderEjs(import.meta.url, "<%- include('../partials/navbar'); %>", {
+          ...resLocals,
+          navPage: 'assessment_instance',
+        })}
         ${resLocals.assessment.type === 'Exam' && resLocals.authz_result.authorized_edit
           ? ConfirmFinishModal({
               instance_questions: resLocals.instance_questions,
               csrfToken: resLocals.__csrf_token,
             })
           : ''}
         ${showTimeLimitExpiredModal ? TimeLimitExpiredModal({ showAutomatically: true }) : ''}
-        ${renderEjs(import.meta.url, "<%- include('../partials/navbar'); %>", {
-          ...resLocals,
-          navPage: 'assessment_instance',
-        })}
+        ${userCanDeleteAssessmentInstance
+          ? RegenerateInstanceModal({ csrfToken: resLocals.__csrf_token })
+          : ''}
+
         <main id="content" class="container">
+          ${userCanDeleteAssessmentInstance
+            ? html`
+                <div class="alert alert-warning alert-dismissible fade show">
+                  You are viewing your own assessment instance and have instructor permissions.
+                  <a
+                    href="#"
+                    role="button"
+                    data-toggle="modal"
+                    data-target="#regenerateInstanceModal"
+                  >
+                    Regenerate this assessment instance
+                  </a>
+                  to pick up any changes to the underlying assessment or if you want to get a fresh
+                  set of questions.
+
+                  <form method="POST" class="ml-auto mr-3 d-none">
```

Comment:
This form is currently hidden with no way to show it. Should it be shown or should it be removed?

Reply:
Good call, it should be removed.